src/main/java/org/generationcp/middleware/service/impl/GermplasmGroupingServiceImpl.java

```diff
@@ -117,8 +118,26 @@ public GermplasmGroup getGroupMembers(final Germplasm founder) {
 	}
 
 	@Override
-	public List<Germplasm> getGroupMembers(final Integer gid) {
-		return this.germplasmDAO.getManagementGroupMembers(gid);
+	public List<Germplasm> getDescendantGroupMembers(final Integer gid, final Integer mgid) {
+		final List<Germplasm> descendantGroupMembers = new ArrayList<>();
+		this.populateDescendantGroupMembers(descendantGroupMembers, gid, mgid);
+		return descendantGroupMembers;
+	}
+
+	private void populateDescendantGroupMembers(final List<Germplasm> descendantGroupMembers, final Integer gid, final Integer mgid) {
+		// Get advanced children
+		final List<Germplasm> maintenanceAndDerivativeChildren = new ArrayList<>();
+		maintenanceAndDerivativeChildren.addAll(this.germplasmDAO.getChildren(gid, 'M'));
+		maintenanceAndDerivativeChildren.addAll(this.germplasmDAO.getChildren(gid, 'D'));
```

Comment:
Refactor `germplasmDao.getChildren` to `getNonGenerativeChildren(final Integer gid, final List<MethodType> filterMethodTypes)`